### DISCUSSION TITLE
is there a component.getStyle() that returns the styles across all media queries?

### DISCUSSION THREAD

**Discussion by glaprida** [GrapesJS/grapesjs#5960] [Q&A]
As far as I can see with simple testing, component.getStyle() right now only returns the styles of the currently active media query.

just by doing
```
editor.on('component:selected',(c)=>{
   console.log(c.getStyle() )
});
```
I can see that if I have an element with base styles (applied on the desktop view) and other styles applied on mobile view; it will only return the styles of the currently active media query; meaning if I'm on desktop view, it will only return those styles... and if I switch to mobile view, it will only return those styles.

Is there a way to get ALL media styles of an element at once? I notice on the docs that the `getStyle()` method allows for `{opts,optsAdd}` but I don't know what those do.

Thanks!

**Comment by artf** [comment]
You can use [`getRules`](https://grapesjs.com/docs/api/css_composer.html#getrules) with a selector or get all of the rules and filter as you wish
